### PR TITLE
Add PullSecretMountPath to ClusterDetails

### LINKED ISSUE #731: Skaffold/Kaniko with ECR
This is a feature request.  I'd like to be able to use Skaffold to create a Kaniko container/pod on KOPS cluster for building container images and then pushing them to an ECR registry.  I'd also like to be able to use an S3 bucket for the build context.

### PR DESCRIPTION
**Pull request by priyawadhwa**
Relates to _in case of new feature, this should point to issue/(s) which describes the feature_

Fixes #731 

**Description**

This will give users the option to specify where the pull secret should
be mounted within the container. This should fix #731 and enable ECR
support.

**User facing changes**

n/a

**Next PRs.**

n/a

**Submitter Checklist**

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [unit tests](../DEVELOPMENT.md#creating-a-pr)
- [ ] Mentions any output changes.
- [ ] Adds documentation as needed: user docs, YAML reference, CLI reference.
- [ ] Adds integration tests if needed.

_See [the contribution guide](../CONTRIBUTING.md) for more details._

Double check this list of stuff that's easy to miss:

- If you are adding [a example to the `examples` dir](https://github.com/GoogleContainerTools/skaffold/tree/master/examples), please copy them to [`integration/examples`](https://github.com/GoogleContainerTools/skaffold/tree/master/integration/examples)
- Every new example added in [`integration/examples` dir](https://github.com/GoogleContainerTools/skaffold/tree/master/integration/examples), should be tested in [integration test](https://github.com/GoogleContainerTools/skaffold/tree/master/integration)

**Reviewer Notes**

- [ ] The code flow looks good. 
- [ ] Unit test added.
- [ ] User facing changes look good.

### CODE REVIEW

File: pkg/skaffold/schema/defaults/defaults.go [hunk 1]
```diff
@@ -210,6 +210,7 @@ func setDefaultClusterTimeout(cluster *latest.ClusterDetails) error {
 }
 
 func setDefaultClusterPullSecret(cluster *latest.ClusterDetails) error {
+	cluster.PullSecretMountPath = valueOrDefault(cluster.PullSecretMountPath, constants.DefaultKanikoSecretMountPath)
```
Comment (tejal29) [inline pkg/skaffold/schema/defaults/defaults.go]: kaniko secret can be optional now. In that case, we need to move this down i guess?

File: pkg/skaffold/schema/defaults/defaults_test.go [hunk 2]
```diff
@@ -111,6 +111,27 @@ func TestSetDefaultsOnCluster(t *testing.T) {
 
 		t.CheckNoError(err)
 		t.CheckDeepEqual(constants.DefaultKanikoSecretName, cfg.Build.Cluster.PullSecretName)
+		t.CheckDeepEqual(constants.DefaultKanikoSecretMountPath, cfg.Build.Cluster.PullSecretMountPath)
```
Comment (tejal29) [inline pkg/skaffold/schema/defaults/defaults_test.go]: Add a test for when the `secret` is optional and make sure `PullSecretMountPath` is not null.

File: pkg/skaffold/schema/defaults/defaults.go (hunk 1 above)
Comment (priyawadhwa) [reply pkg/skaffold/schema/defaults/defaults.go]: This needs to be set for both `PullSecret` and `PullSecretName`, so I put it at the top. If neither of those are set, this will never be used, so it's ok to set it up here.

File: pkg/skaffold/schema/defaults/defaults_test.go (hunk 2 above)
Comment (priyawadhwa) [reply pkg/skaffold/schema/defaults/defaults_test.go]: By "when the secret is optional" do you mean when `PullSecret` and `PullSecretName` are not set?

Comment (tejal29) [reply pkg/skaffold/schema/defaults/defaults_test.go]: yes i think that what @prary accomplished in #2910.

File: pkg/skaffold/schema/defaults/defaults.go (hunk 1 above)
Comment (tejal29) [reply pkg/skaffold/schema/defaults/defaults.go]: umm, so we end up setting defaults for unused field.
Maybe, we should step back.. so, i looked at the bug you pointed, here. I will grab you offline.
